bundle/python/transform.go

```diff
@@ -0,0 +1,129 @@
+package python
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/libraries"
+	"github.com/databricks/databricks-sdk-go/service/compute"
+	"github.com/databricks/databricks-sdk-go/service/jobs"
+)
+
+const NOTEBOOK_TEMPLATE = `# Databricks notebook source
+%python
+{{range .Libraries}}
+%pip install --force-reinstall {{.Whl}}
+{{end}}
+
+from contextlib import redirect_stdout
+import io
+import sys
+sys.argv = [{{.Params}}]
+
+import pkg_resources
+_func = pkg_resources.load_entry_point("{{.Task.PackageName}}", "console_scripts", "{{.Task.EntryPoint}}")
+
+f = io.StringIO()
+with redirect_stdout(f):
+	_func()
+s = f.getvalue()
+dbutils.notebook.exit(s)
+`
+
+// This mutator takes the wheel task and transforms it into notebook
+// which installs uploaded wheels using %pip and then calling corresponding
+// entry point.
+func TransformWheelTask() bundle.Mutator {
+	return &transform{}
+}
+
+type transform struct {
+}
+
+func (m *transform) Name() string {
+	return "python.TransformWheelTask"
+}
+
+func (m *transform) Apply(ctx context.Context, b *bundle.Bundle) error {
+	wheelTasks := libraries.FindAllWheelTasks(b)
+	for _, wheelTask := range wheelTasks {
+		taskDefinition := wheelTask.PythonWheelTask
+		libraries := wheelTask.Libraries
+
+		wheelTask.PythonWheelTask = nil
+		wheelTask.Libraries = nil
+
+		filename, err := generateNotebookWrapper(b, taskDefinition, libraries)
+		if err != nil {
+			return err
+		}
+
+		internalDir, err := b.InternalDir()
+		if err != nil {
+			return err
+		}
+
+		internalDirRel, err := filepath.Rel(b.Config.Path, internalDir)
+		if err != nil {
+			return err
+		}
+
+		parts := []string{b.Config.Workspace.FilesPath}
+		parts = append(parts, strings.Split(internalDirRel, string(os.PathSeparator))...)
+		parts = append(parts, filename)
+
+		wheelTask.NotebookTask = &jobs.NotebookTask{
+			NotebookPath: path.Join(parts...),
+		}
+	}
+	return nil
+}
```

Comment:
This mutator can use an e2e unit test. The net result will be a couple files and transformations to a job so it should be observable without accessing a real workspace.